src/rime/gear/chord_composer.cc

```diff
@@ -104,7 +106,9 @@ ProcessResult ChordComposer::ProcessChordingKey(const KeyEvent& key_event) {
   editing_chord_ = true;
   bool is_key_up = key_event.release();
   if (is_key_up) {
-    if (pressed_.erase(ch) != 0 && pressed_.empty()) {
+    if (finish_when_release_) {
```

Comment:
do we need to check if the released key is tracked as pressed key in the current chord?

consider this scenario:

1. key presses `d, f` forms a chord.
2. when releasing `d`, the chord is finished and cleared. but then, the key `f` is not yet released.
3. if the next key event is `{Release+f}`, since the chord is empty, it's silently ignored as we expect.
4. after moment 2. if user is typing fast and started a new chord `j, k` with a different hand, the key press event `j` could come before `{Release+f}`. our clever program should handle it well, ignoring the slowly releasing `f`.
5. but the current code would see `{Release+f}` as a signal of finishing the next forming chord `j, k`; in the worst case, it could fire an incomplete chord, if the event sequence goes like ... `j, {Release+f}, k` ...

testing if the released key is among `pressed_` should fix the bug at step 5.

Reply:
well I change it, maybe it has solved issue you memtioned.

```
initializing...
message: [0] [deploy] start
message: [0] [deploy] success
ready.
wl
schema: combo_pinyin / 宮保拼音
status: composing
|[ce]
{Release+l}
schema: combo_pinyin / 宮保拼音
status: composing
[ce']|
page: 1  (of size 5)
1. [測]
2.  側
3.  冊
4.  策
5.  廁
s
schema: combo_pinyin / 宮保拼音
status: composing
[ce']|[si]
page: 1  (of size 5)
1. [測]
2.  側
3.  冊
4.  策
5.  廁
{Release+w}
schema: combo_pinyin / 宮保拼音
status: composing
[ce']|[si]
page: 1  (of size 5)
1. [測]
2.  側
3.  冊
4.  策
5.  廁
dj
schema: combo_pinyin / 宮保拼音
status: composing
[ce']|[shi]
page: 1  (of size 5)
1. [測]
2.  側
3.  冊
4.  策
5.  廁
{Release+j}
schema: combo_pinyin / 宮保拼音
status: composing
[ce'shi']|
page: 1  (of size 5)
1. [測試]
2.  側室
3.  側視
4.  測時
5.  策士
```